uiowa/uiowa
# [ITS] Alert fields round 2

Resolves #7058 





# How to test


```
git pull && ddev blt ds --site=its.uiowa.edu && ddev drush @its.local uli /node/add/alert
```

Check that the update hook completes successfully

Title should be hidden, and updated on save based on alert settings.
New "Additional title details" field that affects the alert's title
Category field should be required and allow only 1 value to be selected
Services and Buildings affected fields are select lists allowing multiple values, and the buildings should be prepopulated (there is only one service in prod, but adding more services should allow for more options here as well)

Upon save, the alert title should be set automatically.
Try various combos of including/not including building(s), service(s), and additional details, and see that an accurate title matches each

Test the `isClosed()` method on the Alert Bundle class
`xdebug` works, or you can do something like updating Lines 61+ in `its_core.module.php` with
```
      $output = implode(' - ', array_filter([
        ($entity->isClosed()) ? 'Closed' : 'Open',
        $category,
        $services_buildings,
        $additional_details,
      ]));
```
^ Then create or edit alerts and change the "dates affected" field and see the results in the updated title.
Essentially, if an alert has a definitive end time and that time has passed, then it should show up as "closed," but if it is ongoing, or if the end hasn't passed (whether or not the beginning has passed), it should be considered unclosed.
